thriftsimple/thriftsimple.go

```diff
@@ -0,0 +1,160 @@
+package thriftsimple
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"time"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/getsentry/raven-go"
+	"github.com/reddit/baseplate.go/edgecontext"
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/metricsbp"
+	"github.com/reddit/baseplate.go/secrets"
+	"github.com/reddit/baseplate.go/thriftbp"
+	"github.com/reddit/baseplate.go/tracing"
+	"gopkg.in/yaml.v2"
+)
+
+type BaseplateServerConfig struct {
+	Addr string
+
+	Timeout time.Duration
+
+	Debug bool
+
+	Metrics struct {
+		Namespace string
+		Endpoint  string
+	}
+
+	Secrets struct {
+		Path string
+	}
+
+	Sentry struct {
+		DSN         string
+		Environment string
+		SampleRate  float64
+	}
+
+	Tracing struct {
+		Namespace     string
+		Endpoint      string
+		RecordTimeout time.Duration `yaml:"recordTimeout"`
+		SampleRate    float64
+	}
+}
+
+func (c BaseplateServerConfig) AsConfig() BaseplateServerConfig {
+	return c
+}
+
+func ParseBaseplateServerConfig(path string, cfg interface{}) error {
+	if path == "" {
+		return errors.New("no config path given")
+	}
+
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	if err = yaml.Unmarshal(data, &cfg); err != nil {
+		return err
+	}
+
+	log.Debugf("%#v", cfg)
+	return nil
+}
+func initLogger(debug bool) (log.Level, log.Wrapper) {
+	var logLevel log.Level
+	if debug {
+		logLevel = log.DebugLevel
+	} else {
+		logLevel = log.WarnLevel
```

Comment:
There's also `InfoLevel` between `DebugLevel` and `WarnLevel`, so I'm not sure why when debug is false we skipped that one and go directly to `WarnLevel`.

Reply:
Yeah probably worth fixing!